sdk/core/azure-core/inc/azure/core/http/curl/curl connection_pool.hpp

```diff
@@ -27,6 +27,77 @@ namespace Azure { namespace Core { namespace Test {
 
 namespace Azure { namespace Core { namespace Http {
 
+  /**
+   * @brief The available options to set libcurl ssl options.
+   *
+   * @remark The SDK will map the enum option to libcurl's specific option. See more info here:
+   * https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html
+   *
+   */
+  enum class CurlTransportSSLOptions
+  {
+    Default,
+    AllowBeast,
+    NoRevoke,
+    NoPartialchain,
+    RevokeBesteffort,
+    NativeCa
+  };
+
+  /**
+   * @brief Set the curl connection options like a proxy and CA path.
+   *
+   */
+  struct CurlTransportOptions
+  {
+    /**
+     * @brief The string for the proxy is used without any parsing to libcurl handle.
+     *
+     * @remark No validation for the string is done by the Azure SDK. More about this option:
+     * https://curl.haxx.se/libcurl/c/CURLOPT_PROXY.html.
+     *
+     */
+    std::string proxy;
+    /**
+     * @brief The string for the certificate authenticator is set to libcurl handle directly.
+     *
+     * @remark The Azure SDK will not check if the path is valid or not.
+     *
+     * @remark The default is the built-in system specific. More about this option:
+     * https://curl.haxx.se/libcurl/c/CURLOPT_CAINFO.html
+     *
+     */
+    std::string caInfo;
+    /**
+     * @brief All Http requests will keep the connection channel open to the service.
+     *
+     * @remark The channel might be closed by the server if the server response has an error code.
+     * And a connection won't be re-used if it is abandoned in the middle of an upload or download
```

Comment:
Don't be specific to storage service operations.
```suggestion
     * A connection won't be re-used if it is abandoned in the middle of an operation.
```